assignments/hackyourtemperature/__tests__/app.test.js

```diff
@@ -0,0 +1,68 @@
+import { server } from "../server.js";
+import app from "../app.js";
+import supertest from "supertest";
+
+const request = supertest(app);
+
+
+describe("POST /", () => {
+  it("check if city name is valid", async () => {
+    const response = await request
+    .post('/weather')
+    .send({"cityName": "London"});
+    expect(response.status).toBe(200);
+  });
+
+  it("check if city name is missing", async () => {
+    const response = await request
+    .post('/weather')
+    .send({"cityName": ""});
+    expect(response.status).toBe(400)
+  });
+
+  it("check if city name is wrong", async () => {
+    const response = await request
+    .post('/weather')
+    .send({"cityName": "Londondondon"});
+    expect(response.body).toEqual({ error: "API error" });
```

Comment:
In case the City is wrong, we might want to answer with more explanations. From this, a good practice would be to:

  - update this line to match what we would our server to answer
    ```suggestion
        expect(response.status).toBe(404);
        expect(response.body).toEqual({ error: "City Londondondon not found" });
    ```
  - Run the tests **at that point the tests should be red**
  - Change your code to answer with 404 and the new message in case the city do not exist
  - Run the test
  - Until the tests is not green, try to update your code and make it green
  
  This is a suggestion to push your code one step further